File: Semester 4/Programming and Projection Enironments/Proiect_MPP/src/main/java/root/proiect_mpp/domain/people/Person.java
package root.proiect_mpp.domain.people;

import root.proiect_mpp.domain.Entity;

public abstract class Person implements Entity<Integer> {
    private int id;
    private String firstName;
    private String lastName;

    // Class Constructors //

    public Person() {
        this.id = 0;
        this.firstName = "";
        this.lastName = "";
    }

    public Person(String firstName, String lastName) {
        this.id = 0;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public Person(int id, String firstName, String lastName) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    // Getters & Setters //

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    // toString & other functions //

    @Override
    public String toString() {
        return  "id=" + id +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'';
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/model/people/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model.people
{
    [Serializable]
    public class Person : Entity<int>
    {
        private string firstName;
        private string lastName;

        // Class Constructors //

        public Person() : base(0)
        {
            this.firstName = "";
            this.lastName = "";
        }

        public Person(string firstName, string lastName) : base(0)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public Person(int id, string firstName, string lastName) : base(id)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        // Getters & Setters //

        public string FirstName
        {
            get { return this.firstName; }
            set { this.firstName = value; }
        }

        public string LastName
        {
            get { return this.lastName; }
            set { this.lastName = value; }
        }

        // toString & other functions //

        public override string ToString() 
        {
            return "id=" + base.ID +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'';
        }
    }
}
